docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,895 @@
+    default: conf2
+  - targetRef:
+      kind: MeshHTTPRoute
+      name: slow-backend-route
+      namespace: backend-ns
+    default: conf3
+```
+
+When merging we will merge `$conf1` and `$conf2` for MeshService with `$conf3` for `Mesh*Route`. 
+
+```yaml
+resourceRules:
+  backend.backend-ns: 
+    conf: $conf1
+  frontend.backend-ns:
+    conf: $conf2
+  backend-route.backend-ns:
+    conf: merge($conf1, $conf2, $conf3)
+```
+
+Which makes it problematic when applying conf since we are applying policy per outbound, and we have merged config for 
+multiple outbounds. 
+
+Possible solutions:
+
+1. We can deprecate and then remove possibility to have multiple `spec.to[]` items in `Mesh*Route` policy to make it referencable. 
+(this is basically step into removing `to` section altogether, why do we need it if we cannot put multiple items in it).
+2. When merging we should only merge confs for given resource, don't treat `Mesh*Route` as subtype of `MeshService` and then
+when applying configuration to resource we should be more mindful on which outbound the route is configured and configure 
+it based on conf for that MeshService
+
+**Decision outcome**
+
+We've decided to go with option 1, which is to deprecate and later on remove possibility to put multiple items in `spec.to[]`
+section of `Mesh*Route`. This will align `Mesh*Route` design with GatewayAPI. In the future we should also discuss removing 
+`spec.to[]` section entirely leaving only `spec.targetRef` section in `Mesh*Route`.
+
+#### Overriding configuration
+
+For policies on namespace this is as simple as creating consumer policy that targets MeshService or `Mesh*Route`:
+(This can be useful when we consume MeshService from other zone)
+
+```yaml
+# Consumer MeshTimeout targeting whole MeshService
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend-service
+  namespace: frontend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+      namespace: backend-ns
+    default: $conf1
+---
+# Consumer MeshTimeout targeting producer route
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend-route
+  namespace: frontend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshHTTPRoute
+      name: route-to-backend
+      namespace: backend-ns
+    default: $conf2
+```
+
+After merging conf we will get:
+
+```yaml
+resourceRules:
+  backend.backend-ns: 
+    conf: $conf1
+  backend-route.backend-ns:
+    conf: merge($conf1, $conf2)
+```
+
+Configuration priority (from most to least important): 
+1. consumer policy targeting `Mesh*Route`
+2. producer policy targeting `Mesh*Route`
+3. consumer policy targeting MeshService
+4. producer policy targeting MeshService
+
+We always prioritize policies targeting `Mesh*Route` than the one targeting MeshService because route can have more specific
+requirements than a whole MeshService
```

Comment:
How do you prioritise with a route on the consumer side vs on the producer side?

Reply:
I've updated this information and updated merging order with consumer/producer routes